[Python Test] Added TC_APPLAUNCHER_3_5 with CatalogList type and value validation

Rajashreekalmane opened the pull request:
#### Summary
Added Python test script for TC_APPLAUNCHER_3_5 by adding strict validation for the CatalogList attribute in the AppLauncher cluster.


Test Plan PR Reference : https://github.com/CHIP-Specifications/chip-test-plans/pull/5383 

#### Related issues


#### Testing

1. DUT Launch Command: ./chip-tv-app

2. Test Script Execution Command:
python3 TC_APPLAUNCHER_3_5.py --commissioning-method on-network --discriminator 3840 --passcode 20202021 --storage-path admin_storage.json --endpoint 1

**Result:**
Test case successfully passed and validated manually.

#### Readability checklist

Rajashreekalmane committed:
Added new python test case TC_APPLAUNCHER_3_5

Rajashreekalmane committed:
minor changes

Rajashreekalmane committed:
Update TC_APPLAUNCHER_3_5.py

Rajashreekalmane committed:
Update TC_APPLAUNCHER_3_5.py

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

restyled-commits committed:
Restyled by autopep8

Rajashreekalmane committed:
minor  changes

Rajashreekalmane committed:
minor update

Rajashreekalmane committed:
removed YAML Script

raul-marquez-csa commented on src/python_testing/TC_APPLAUNCHER_3_5.py:
```diff
@@ -0,0 +1,76 @@
+            attribute=attributes.CatalogList
+        )
+        # Confirm it's a list
+        asserts.assert_true(
```
Please swap for asserts.assert_is_instance

raul-marquez-csa commented on src/python_testing/TC_APPLAUNCHER_3_5.py:
```diff
@@ -0,0 +1,76 @@
+        attributes = cluster.Attributes
+
+        # Step 1: Commissioning (already completed)
+        self.step(1)
```
Test steps in script vs test plan don't match.
For commissioning step you can use something like:
`            TestStep("precondition", "DUT commissioned and preconditions", is_commissioning=True),`
`self.step("precondition")`
This is common among other test scripts.
This way Step 1, the only step in the test plan, can match step 1 from the script.

raul-marquez-csa commented on src/python_testing/TC_APPLAUNCHER_3_5.py:
```diff
@@ -0,0 +1,76 @@
+
+        logging.info(f"CatalogList: {catalog_list}")
+
+        # Validate each entry is a valid CSA-issued Vendor ID (uint16)
+        for vendor_id in catalog_list:
+            asserts.assert_greater(len(catalog_list), 0, "CatalogList attribute is empty or not present")
```
You're looping the catalog_list for verification of vendor_id, but the assertion is for the length of the catalog_list, it's a copy of the assertion above this code block.

For the assertion you can use `assert_valid_uint16` from `matter_testing_infrastructure/matter/testing/matter_asserts.py`

Rajashreekalmane committed:
resolved review comments

Rajashreekalmane commented:
@raul-marquez-csa All review comments have been addressed:
1. Replaced type checks with assert_is_instance
2. Updated commissioning step to "precondition" for consistency
3. Removed unused imports and duplicate assertions
4. Used assert_valid_uint16 for Vendor ID validation

raul-marquez-csa reviewed (approved):
LGTM

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

restyled-commits committed:
Restyled by isort

Rajashreekalmane committed:
Update TC_APPLAUNCHER_3_5.py

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

Rajashreekalmane committed:
Remove commented CI test arguments

Removed commented-out CI test arguments for clarity.

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

Rajashreekalmane committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5

ukatira committed:
Merge branch 'master' into TC-APPLAUNCHER-3.5
